search/engine.cpp

```diff
@@ -138,42 +139,43 @@ weak_ptr<ProcessorHandle> Engine::Search(SearchParams const & params)
 
 void Engine::SetLocale(string const & locale)
 {
-  PostMessage(Message::TYPE_BROADCAST, [this, locale](Processor & processor)
-              {
-                processor.SetPreferredLocale(locale);
-              });
+  PostMessage(Message::TYPE_BROADCAST,
+              [locale](Processor & processor) { processor.SetPreferredLocale(locale); });
 }
 
 void Engine::ClearCaches()
 {
-  PostMessage(Message::TYPE_BROADCAST, [this](Processor & processor)
-              {
-                processor.ClearCaches();
-              });
+  PostMessage(Message::TYPE_BROADCAST, [](Processor & processor) { processor.ClearCaches(); });
 }
 
 void Engine::LoadCitiesBoundaries()
 {
   PostMessage(Message::TYPE_BROADCAST,
-              [this](Processor & processor) { processor.LoadCitiesBoundaries(); });
+              [](Processor & processor) { processor.LoadCitiesBoundaries(); });
+}
+
+void Engine::LoadCountriesTree()
+{
+  PostMessage(Message::TYPE_BROADCAST,
+              [](Processor & processor) { processor.LoadCountriesTree(); });
 }
 
 void Engine::OnBookmarksCreated(vector<pair<bookmarks::Id, bookmarks::Doc>> const & marks)
 {
   PostMessage(Message::TYPE_BROADCAST,
-              [this, marks](Processor & processor) { processor.OnBookmarksCreated(marks); });
+              [marks](Processor & processor) { processor.OnBookmarksCreated(marks); });
```

Comment:
`[&marks]`?

Reply:
Do you really think it's a good idea to pass some value on another thread by reference?

Reply:
It's another thread? No. It's not a good idea.